pkg/uefi/meregion.go

```diff
@@ -0,0 +1,238 @@
+// Copyright 2018 the LinuxBoot Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package uefi
+
+import (
+	"bytes"
+	"encoding/binary"
+	"encoding/hex"
+	"fmt"
+	"log"
+)
+
+// ME Partition parsing, the goal is to spot a padding in the ME Region
+// after the ME partitions so that this region can be shrunk in the IFD.
+//
+// ME partition informations from http://me.bios.io/ME_blob_format
+
+// MEFTPSignature is the sequence of bytes that an ME Flash Partition
+// Table is expected to start with.
+var (
+	MEFTPSignature = []byte{0x24, 0x46, 0x50, 0x54}
+)
+
+const (
+	// MEFTPSignatureLength represents the size of the ME FTP signature
+	MEFTPSignatureLength = 4
+	// MEPartitionTableEntryLength is the size of a partition table entry
+	MEPartitionTableEntryLength = 32
+)
+
+// MEFPT is the main structure that represents an ME Flash Partition Table.
+type MEFPT struct {
+	// Holds the raw buffer
+	buf []byte
+
+	PartitionCount    uint32
+	PartitionMapStart int
+	Entries           []MePartitionEntry
+	//Metadata for extraction and recovery
+	ExtractPath string
+}
+
+// MePartitionEntry is an entry in FTP
+type MePartitionEntry struct {
```

Comment:
Perhaps, for consistency with the other names, this should be called `MEPartitionEntry` (capitalizing the E).

Reply:
Right, also more consistent with go conventions for initials in names